torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2897,7 +2901,7 @@ def _post_backward_hook(self, param: Parameter, *unused: Any) -> None:
                     # reduce_dtype matches the param dtype.
                     param.grad.data = param.grad.data.to(self.mixed_precision.reduce_dtype)
 
-                if self.gradient_predivide_factor > 1:
+                if self.gradient_predivide_factor > 1 and self.communication_hook is None:
                     # Average grad by world_size for consistency with PyTorch DDP.
```

Comment:
(This is prior to the PR) The comment does not seem to match what we do here? `gradient_predivide_factor` might or might not be the `world_size`?

Reply:
Agreed, the process of pre + post division together is what leads overall to the averaging by world size. Would be good to send a separate PR updating the docs to reflect this.

Reply:
Sounds good, I will take a look and prepare a separate PR